applications/StructuralMechanicsApplication/custom_strategies/custom_strategies/eigensolver_strategy.hpp

```diff
@@ -402,7 +405,7 @@ class EigensolverStrategy
         if (rModelPart.NumberOfMasterSlaveConstraints() != 0) {
             this->pGetBuilderAndSolver()->ApplyConstraints(pScheme, rModelPart, rMassMatrix, b);
         }
-        this->ApplyDirichletConditions(rMassMatrix, 1.0);
```

Comment:
Because of the reasons stated in the comments, this change would produce wrong results for a `BlockBuilderAndSolver`